packages/plugins/typescript/generic-sdk/src/visitor.ts

```diff
@@ -91,9 +104,7 @@ export class GenericSdkVisitor extends ClientSideBaseVisitor<RawGenericSdkPlugin
 
     const documentNodeType = this.config.documentMode === DocumentMode.string ? 'string' : 'DocumentNode';
     const resultData = this.config.rawRequest ? 'ExecutionResult<R, E>' : 'R';
-    const returnType = usingObservable
-      ? `Promise<${resultData}> & Observable<${resultData}>`
-      : `Promise<${resultData}>`;
+    const returnType = `Promise<${resultData}> | ${usingObservable ? 'Observable' : 'AsyncIterable'}<${resultData}>`;
```

Comment:
why changing `Promise & ...` to `Promise | ...` ?

Reply:
I think `&` is wrong because result can be `Promise` or stream. `&` makes it to have both signatures which is wrong actually, right?